### PR TITLE
feat: add type declaration

### PR DESCRIPTION
**Pull request by Pduhard**
add type delcaration for CallbackManager, ConfigStorage, Debug, Mainloop and ClassUtils

### CODE REVIEW

File: types/helpers/class-utils.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const getter: () => any;
+export const setter: (value: any) => void;
+export const applyProperties: (instance: any, params: any) => void;
```
Comment (jbghoul-wanadev) [inline types/helpers/class-utils.d.ts:3]: export default with the three of those ?

File: types/lib/debug.d.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import Class from "abitbol";
+import type CallbackManager from "./callback-manager";
+
+// TODO throw errors for console.error()
```
Comment (jbghoul-wanadev) [inline types/lib/debug.d.ts:4]: what should we do ?

File: types/helpers/class-utils.d.ts (hunk 1 above)
Comment (Pduhard) [reply types/helpers/class-utils.d.ts:3]: https://github.com/wanadev/obsidian-core/pull/112/commits/1dc7fb53727793d5f9faeb2d7214675eef036a94

File: types/lib/debug.d.ts (hunk 2 above)
Comment (Pduhard) [reply types/lib/debug.d.ts]: https://github.com/wanadev/obsidian-core/pull/112/commits/49b759b20f9de1a744662002f63fc2460f559e05

File: types/helpers/class-utils.d.ts (hunk 1 above)
Comment (jbghoul-wanadev) [reply types/helpers/class-utils.d.ts:3]: I meant adding a new line at the end:

```
export const getter: () => any;
export const setter: (value: any) => void;
export const applyProperties: (instance: any, params: any) => void;

export default {
     getter,
     setter,
     applyProperties
}
```

**Review by jbghoul-wanadev — APPROVED** [review]

**Pull request merged**